pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -713,8 +714,19 @@ func PodFitsResources(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 	}
 
 	var podRequest *schedulercache.Resource
+	// No extended resources should be ignored by default unless specified via
+	// extender configuration.
+	ignoredExtendedResources := sets.NewString()
+	// Missing extended resources will be ignored by default unless specified
+	// in predicate metadata. Note that kube-scheduler sets this flag to false
+	// in predicate metadata.
+	ignoreMissingExtendedResources := true
```

Comment:
Thanks for looking into this over the weekend!

Yes, this is a backward incompatible change but I haven't come up with a promising solution yet.

In `PodFitsResources`, we cast `algorithm.PredicateMetadata` to `predicate.predicateMetadata` and then use the metadata, including `ignoreMissingExtendedResources` in this PR.
https://github.com/kubernetes/kubernetes/blob/0394ffba6de860ece173a9a485d3518d3a8d37f2/pkg/scheduler/algorithm/predicates/predicates.go#L716-L717

This is how the predicate is invoked in pod admission on node. https://github.com/kubernetes/kubernetes/blob/0394ffba6de860ece173a9a485d3518d3a8d37f2/pkg/kubelet/lifecycle/predicate.go#L80 Here we pass `nil` as the predicate metadata, which means we use the defaults in pod admission on node. That's why I set  `ignoreMissingExtendedResources` default to `true`.

Ideally, kubelet should create a predicate metadata with `ignoreMissingExtendedResources` set to `true` and pass it to the predicate. But this is not possible because of the type cast in the predicate function.

I think the type cast should be removed, and the metadata should be accessed via the `algorithm.PredicateMetadata` interface. kube-scheduler/node/clusterautoscalar can provide their own metadata implementation. But this will be a larger change. Maybe I'm missing some thing. WDYT?

Reply:
I see. So, for now please add a comment and explain that this default value is chosen so that the kubelet code works fine and add the details.
My only concern right now is that this is backward incompatible from the kubelet side. It is however, backward compatible on the scheduler side, because if scheduler's policy config does not exists or misses this field, the default value of the field will be false and overrides the default value provided here.

Reply:
Done.

Besides scheduler and kubelet, cluster autoscaler and damonset controller also use this predicate. But I don't see any negative impacts from this change. This is also the behavior before 1.8 https://github.com/kubernetes/kubernetes/issues/53616.

@vishh @jiayingz

Reply:
I created `algorithm.PredicateMetadataProducer` in kubelet using `predicates.NewPredicateMetadataFactory()`.

It turned out that `NewPredicateMetadataFactory`'s `podLister` argument is not used at all so I can just use this function in kubelet.